### PR TITLE
fix: Split expense lint fix

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=I6AXM4O)
### <samp>🤖 Generated by Copilot at 9d8fc5b</samp>

This pull request adds cost center support for split expenses and improves the type safety and readability of the split-expense feature. It modifies the `SplitExpense`, `FileTransaction`, and `SplitExpenseForm` interfaces, and updates the `split-expense.page.ts` and `split-expense.service.ts` files accordingly. It also removes some unused code and imports.

### <samp>🤖 Generated by Copilot at 9d8fc5b</samp>

> _`SplitExpenseForm`:_
> _A new interface for spring_
> _Cleaning up the code_

### <samp>🤖 Generated by Copilot at 9d8fc5b</samp>

*  Add `SplitExpenseForm` interface to define the form group for each split expense item ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-985911cc21b108329ba00be403cabf7b7c17c97970e80fdee2a57253aaf60208R1-R7))
*  Import and use `FileObject` model as a type for the `files` property of the `FileTransaction` interface ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-7a3d072518be859cc7f9c790b69a0a69820ed1aa2a92063fab54ae124cbf0bf0L1-R6), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-bab29bd73c5e5f43534ff8d9b9bc4a680f26d582c82c182ec8cc0b0586aacfd4L41-R41), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L59-R76), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L257-R267), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L277-R286), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L372-R381))
*  Add `cost_center` property to the `SplitExpense` interface and import and use `CostCenter` model as its type ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-a6689d4dac4a99aa1eb245f7a17eae73c453eedf1f1cbdc7127d797a119990c3R1-R2), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-a6689d4dac4a99aa1eb245f7a17eae73c453eedf1f1cbdc7127d797a119990c3R39), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L117-R130), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L156-R163), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L192-R203), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L204-R220), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L219-R228), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L440-R453), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L556-R580), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L647-R673))
*  Remove unused imports and reorder imports in `split-expense.page.ts` file ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L1-R8))
*  Remove `OnInit` interface from `SplitExpensePage` class as it was not implemented ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878R34-R42), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L40-R49))
*  Change `txnFields` property of `SplitExpensePage` class to be of type `Partial<ExpenseFieldsObj>` and import and use `ExpenseFieldsObj` model as its type ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878R34-R42), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L47-R56), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L498-R525), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L580-R604))
*  Change types of `costCenters$`, `transaction`, `fileObjs`, `fileUrls`, `selectedCCCTransaction`, `splitExpenseTxn`, and `completeTxnIds` properties of `SplitExpensePage` class to match the models imported from the `core/models` folder ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L59-R76), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L73-R82), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L83-R92))
*  Remove empty `ngOnInit` method from `SplitExpensePage` class and add explicit return types and parameter types to `goBack`, `onChangeAmount`, and `onChangePercentage` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L117-R130), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L156-R163))
*  Add explicit return types and parameter types to `getTotalSplitAmount`, `setUpSplitExpenseBillable`, `setUpSplitExpenseTax`, and `generateSplitEtxnFromFg` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L192-R203), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L204-R220), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L219-R228))
*  Add explicit return types and parameter types to `uploadNewFiles` and `uploadFiles` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L257-R267), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L277-R286))
*  Add explicit return types and parameter types to `getCategoryList`, `createAndLinkTxnsWithFiles`, `toastWithCTA`, `toastWithoutCTA`, and `showSuccessToast` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L290-R299), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L296-R306), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L322-R331), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L338-R347), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L348-R357))
*  Add explicit return types and parameter types to `getAttachedFiles`, `showSplitExpenseViolations`, and `handleSplitExpensePolicyViolations` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L372-R381), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L381-R390), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L394-R408))
*  Add explicit return types and parameter types to `save` and `getActiveCategories` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L408-R418), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L492-R509))
*  Add explicit return types and parameter types to `ionViewWillEnter`, `setValuesForCCC`, `setAmountAndCurrency`, `customDateValidator`, `add`, and `remove` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L498-R525), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L580-R604), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L621-R650), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L647-R673), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L654-R683))
*  Add explicit return types and parameter types to `splitEvenly`, `setEvenSplit`, and `split` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L671-R700), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L687-R716), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L712-R741))
*  Set `saveSplitExpenseLoading` property to `true` to indicate the loading state in `save` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L440-R453))
*  Cast values from `activatedRoute.snapshot.params` object to match the types of the models imported from the `core/models` folder in `ionViewWillEnter` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L498-R525))
*  Cast values from `splitExpensesFormArray` and `this.txnFields` to match the types of the models imported from the `core/models` folder in various methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L192-R203), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L204-R220), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L219-R228), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L421-R440), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L440-R453), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L451-R468), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L515-R539))
*  Use implicit types inferred from the `forkJoin` operator and the `getbyId` method in various methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L306-R315), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L315-R324), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L451-R468), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L483-R500), [link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L515-R539))
*  Disable eslint checks for the unused `data` variable of the `onWillDismiss` callback in `showSplitExpenseViolations` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2246/files?diff=unified&w=0#diff-e53ec269f6762f843707cfbf61d723efbc80075ec0e590ccdbf2b7a642a7a878L381-R390))

## Clickup
https://app.clickup.com/t/85ztk92au

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 1]
```diff
@@ -56,21 +65,21 @@ export class SplitExpensePage implements OnInit {
 
   categories$: Observable<OrgCategoryListItem[]>;
 
-  costCenters$: Observable<any>;
+  costCenters$: Observable<CostCenters[]>;
```
Comment (Chethan-Fyle) [inline src/app/fyle/split-expense/split-expense.page.ts:68]: Is this intended? Or was this supposed to be `CostCenter[]`?

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 2]
```diff
@@ -335,7 +344,7 @@ export class SplitExpensePage implements OnInit {
     });
   }
 
-  toastWithoutCTA(toastMessage, toastType, panelClass) {
+  toastWithoutCTA(toastMessage: string, toastType: 'success' | 'failure' | 'information', panelClass: string): void {
```
Comment (Chethan-Fyle) [inline src/app/fyle/split-expense/split-expense.page.ts:347]: Define and use an enum for toast type.

**Review by Chethan-Fyle — COMMENTED** [review]
few minor comments.

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 3]
```diff
@@ -31,20 +31,29 @@ import { CustomInput } from 'src/app/core/models/custom-input.model';
 import { FileObject } from 'src/app/core/models/file-obj.model';
 import { LaunchDarklyService } from 'src/app/core/services/launch-darkly.service';
 import { ProjectsService } from 'src/app/core/services/projects.service';
+import { ExpenseFieldsObj } from 'src/app/core/models/v1/expense-fields-obj.model';
+import { CostCenter, CostCenters } from 'src/app/core/models/v1/cost-center.model';
```
[the remark this answers is not in the document]
Comment (suyashpatil78) [reply src/app/fyle/split-expense/split-expense.page.ts:35]: CostCenter has a structure
`CostCenter {
  id: number;
  created_at: Date;
  updated_at: Date;
  name: string;
  code: string;
  active: boolean;
  org_id: string;
  description: string;
}`
while `CostCenters` has `{label: string; value: CostCenter}`

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 4]
```diff
@@ -618,10 +644,10 @@ export class SplitExpensePage implements OnInit {
     }
   }
 
-  add(amount?, currency?, percentage?, txnDt?) {
+  add(amount?: number, currency?: string, percentage?: number, txnDt?: string | number | Date | dayjs.Dayjs): void {
```
[the remark this answers is not in the document]
Comment (suyashpatil78) [reply src/app/fyle/split-expense/split-expense.page.ts:647]: Corrected it!

**Review by Chethan-Fyle — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]

**Pull request merged**